h2o-py/h2o/frame.py

```diff
@@ -1280,34 +1284,63 @@ def drop(self, index, axis=1):
           H2OFrame with the respective dropped columns or rows. Returns a new H2OFrame.
         """
         if axis == 1:
-            if is_type(index, str):
-                index = self.names.index(index)
+            if not isinstance(index,list):
+                #If input is a string, i.e., "C1":
+                if is_type(index, str):
+                    #Check if index is an actual column(s) in the frame
+                    assert set([index]).issubset(self.names) == True, "Column(s) selected to drop are not in " \
+                                                                          "original frame: %r" % index
+                    index = self.names.index(index)
+                #If input is an int indicating a column index, i.e., 3:
+                elif is_type(index, int):
+                    #Check if index is an actual column index in the frame
+                    assert index < self.ncol, "Column index selected to drop is not part of the frame: %r" % index
+                    assert index >= 0, "Column index selected to drop is not positive: %r" % index
+
                 fr = H2OFrame._expr(expr=ExprNode("cols", self, -(index + 1)), cache=self._ex._cache)
                 fr._ex._cache.ncols -= 1
                 fr._ex._cache.names = self.names[:index] + self.names[index + 1:]
                 fr._ex._cache.types = {name: self.types[name] for name in fr._ex._cache.names}
                 return fr
-            if all(isinstance(item, int) for item in index):
-                for i in range(len(index)):
+
+            elif isinstance(index,list):
+                #If input is an int array indicating a column index, i.e., [3] or [1,2,3]:
+                if all(isinstance(item, int) for item in index):
```

Comment:
again, this is much easier to write as `is_type(item, [int])`, or even better `is_type(item, [BoundInt(-self.ncol, self.ncol-1)]`